design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,527 @@
+    checkCompanyRoleName(name: String!): Boolean  @doc(description: "Returns boolean value of validation whether provided Role name is available.") 
+    companyHierarchy: CompanyHierarchyOutput  @doc(description: "Returns the complete data about Company structure.") 
+    companyTeam(id: Int!): CompanyTeam  @doc(description: "Returns Company Team data by ID.") 
+}
+
+type Company @doc(description: "Company entity output data schema.") {
+    id: Int @doc(description: "Company ID.")
+    company_name: String @doc(description: "Company name.")
+    company_email: String @doc(description: "Company email address.")
+    legal_name: String @doc(description: "Company legal name.")
+    vat_tax_id: String @doc(description: "Company VAT/TAX ID.")
+    reseller_id: String @doc(description: "Company re-seller ID.")
+    legal_address: CompanyLegalAddress @doc(description: "An object containing Company legal address data.")
+    company_admin: CompanyAdmin @doc(description: "An object containing information about Company Administrator.")
+    sales_representative: CompanySalesRepresentative @doc(description: "An object containing information about Company representative.")
+	payment_methods: [String] @doc(description: "List of payment methods awailable for a Company.")
+}
+
+type CompanyLegalAddress @doc(description: "Company legal address output data schema.") {
+    street: [String] @doc(description: "An array of strings that defines the Company's street address.")
+    city: String @doc(description: "City name.")
+    region: CompanyAddressRegion @doc(description: "An object containing region data for the Company.")
+    country_id: String @doc(description: "Country ID.")
+    country_name: String @doc(description: "Country name.")
+    postcode: String @doc(description: "ZIP/postal code.")
+    telephone: String @doc(description: "Company's phone number.")
+}
+
+type CompanyAdmin @doc(description: "Company Administrator (Customer with corresponding privileges) output data schema.") {
+    id: Int @doc(description: "Company Administrator's ID.")
+    email: String @doc(description: "Company Administrator email address.")
+    firstname: String @doc(description: "Company Administrator first name.")
+    lastname: String @doc(description: "Company Administrator last name.")
+    job_title: String @doc(description: "Company Administrator job title.")
+    gender: Int @doc(description: "Company Administrator gender.")
+}
+
+type CompanyAddressRegion @doc(description: "Output data schema for a region data in Company address.") {
+    region_id: Int @doc(description: "Unique region identifier. Required for certain countries.")
+    region: String @doc(description: "State or province name.")
+    region_code: String @doc(description: "Region code.")
+}
+
+type CompanySalesRepresentative @doc(description: "Company sales representative information output data schema.") {
```

Comment:
Is this a reference to an existing customer or just some data on the level of `Company` record? If it can be converted into reference to existing `Customer` - it would be better I think.

Reply:
This is a reference to admin customer which doesn't/shouldn't have object on storefront